src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -516,8 +516,18 @@ class WalletService(
                     .switchIfEmpty {
                         Mono.error(WalletSessionMismatchException(session.sessionId, walletId))
                     }
-                    .filter { it.status == WalletStatusDto.INITIALIZED.value }
-                    .switchIfEmpty { Mono.error(WalletConflictStatusException(walletId)) }
+                    .flatMap {
+                        when (it.status) {
+                            WalletStatusDto.INITIALIZED.value -> Mono.just(it)
+                            else ->
+                                Mono.error(
+                                    WalletConflictStatusException(
+                                        walletId,
+                                        WalletStatusDto.valueOf(it.status)
+                                    )
+                                )
+                        }
+                    }
```

Comment:
invoke wallet `expectInStatus` function here too

Reply:
I miss that, thanks 🚀